frontend/src/features/public-form/components/FormPaymentPage/stripe/components/DownloadReceiptBlock.tsx

```diff
@@ -1,46 +1,116 @@
+import { useMemo } from 'react'
 import { BiDownload } from 'react-icons/bi'
+import { Box, Divider, Stack, Text } from '@chakra-ui/react'
+import { format } from 'date-fns'
+
+import { PaymentType, ProductItem } from '~shared/types'
+import { centsToDollars } from '~shared/utils/payments'
 
 import { useToast } from '~hooks/useToast'
 import Button from '~components/Button'
 
 import { getPaymentInvoiceDownloadUrl } from '~features/public-form/utils/urls'
 
-import { GenericMessageBlock } from './GenericMessageBlock'
-
 type DownloadReceiptBlockProps = {
   formId: string
   submissionId: string
   paymentId: string
+  amount: number
+  products: ProductItem[]
+  paymentType?: PaymentType
+  name: string
+  paymentDate?: Date
+}
+
+const PaymentSummaryItem = ({
+  title,
+  input,
+}: {
+  title: string
+  input: string
+}): JSX.Element => {
+  return (
+    <Stack
+      direction={{ base: 'column', md: 'row' }}
+      spacing={{ base: 0, md: '1.5rem' }}
+    >
+      <Text textStyle="body-2" width="6.5rem" color="#848484">
+        {title}
+      </Text>
+      <Text textStyle="body-2" color="#474747">
+        {input}
+      </Text>
+    </Stack>
+  )
 }
 
 export const DownloadReceiptBlock = ({
   formId,
   submissionId,
   paymentId,
+  amount,
+  products,
+  paymentType,
+  name,
+  paymentDate,
 }: DownloadReceiptBlockProps) => {
   const toast = useToast({ status: 'success', isClosable: true })
 
+  // Extract selected product names for payment by products
+  const extractProductNames = (products: ProductItem[]) => {
```

Comment:
/nit This function is already pure, we could also extract this function to be out of `<DownloadReceiptBlock />` as this function will be recreated whenever this component is re-rendered.

This will be ultra-nit, Might also be terser in more fp-style  😆 
```js
const getProductNames = (items: ProductItem[]) => products
  .filter(product => product.selected)
  .map(product => `${product.quantity}x ${product.data.name}`)
  .join(', ')
```

Reply:
[2353977](https://github.com/opengovsg/FormSG/pull/6591/commits/23539777cf1a763a9cb3d35523dc85d0063015f7)